zerver/webhooks/circleci/view.py

```diff
@@ -26,22 +26,103 @@ def api_circleci_webhook(request: HttpRequest, user_profile: UserProfile,
     return json_success()
 
 def get_subject(payload: Dict[str, Any]) -> str:
-    return CIRCLECI_TOPIC_TEMPLATE.format(repository_name=payload['reponame'])
+    repository_name = payload["reponame"]
+    return f"{repository_name}"
+
+def get_commit_range_info(payload: Dict[str, Any]) -> str:
+    commits = payload["all_commit_details"]
+    num_commits = len(commits)
+
+    if num_commits == 1:
+        commit_id = commits[0]["commit"][:10]
+        commit_url = commits[0]["commit_url"]
+        return f"- **Commit:** [{commit_id}]({commit_url})"
+
+    vcs_provider = payload["user"]["vcs_type"]  # Same as payload["why"]?
+    first_commit_id = commits[0]["commit"]
+    shortened_first_commit_id = first_commit_id[:10]
+    last_commit_id = commits[-1]["commit"]
+    shortened_last_commit_id = last_commit_id[:10]
+    if vcs_provider == "github":
+        # Then use GitHub's commit range feature to form the appropriate url.
+        vcs_url = payload["vcs_url"]
+        commit_range_url = f"{vcs_url}/compare/{first_commit_id}...{last_commit_id}"
+        return f"- **Commits ({num_commits}):** [{shortened_first_commit_id} ... {shortened_last_commit_id}]({commit_range_url})"
+    else:
+        # BitBucket doesn't have a good commit range url feature like GitHub does.
+        # So let's just show the two separately.
+        # https://community.atlassian.com/t5/Bitbucket-questions/BitBucket-4-14-diff-between-any-two-commits/qaq-p/632974
```

Comment:
We should test GitLab too.

Reply:
Actually, CircleCI doesn't seem to support GitLab yet. Ref: https://ideas.circleci.com/ideas/CCI-I-248